packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -57,7 +57,7 @@ export class B3Propagator implements HttpTextPropagator {
       // We set the header only if there is an existing sampling decision.
       // Otherwise we will omit it => Absent.
       if (spanContext.traceFlags !== undefined) {
-        setter(carrier, X_B3_SAMPLED, Number(spanContext.traceFlags));
+        setter(carrier, X_B3_SAMPLED, String(spanContext.traceFlags));
```

Comment:
In the future, this assumes that a direct mapping of trace flags to b3 sampled is achievable by just doing toString, but if another bit is added to trace flags this assumption might be broken.
```suggestion
        setter(carrier, X_B3_SAMPLED, (TraceFlags.SAMPLED & parentContext.traceFlags) === TraceFlags.SAMPLED ? '1' : '0');
```

Reply:
sure done